File: 20-valid-parentheses/valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char>st;
        for(int i=0;i<s.size();i++)
        {
            if(s[i]=='(' or s[i]=='{' or s[i]=='[')
            {
                st.push(s[i]);
            }
            else
            {
                if(!st.empty() and ((st.top()=='(' and s[i]==')') or  (st.top()=='{' and s[i]=='}') or  (st.top()=='[' and s[i]==']')))
                {
                    st.pop();
                }
                else
                {
                    return false;
                }
            }
        }
       return st.empty(); 
    }
};

File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        st=[]
        for i in s:
            if i=="{" or i=="[" or i=="(":
                st.append(i)
            else:
                if len(st)==0:
                    return False
                if i=="}" and st[-1]=="{" or i=="]" and st[-1]=="[" or i==")" and st[-1]=="(":
                    st.pop()
                else:
                    return False
                    break
        if len(st)==0:
            return True
        else:
            return False
        